addons/pos_restaurant/tests/test_frontend.py

```diff
@@ -4,46 +4,64 @@
+from odoo.addons.base.tests.common import HttpCaseWithUserDemo
+
 
 @odoo.tests.tagged('post_install', '-at_install')
-class TestFrontend(odoo.tests.HttpCase):
-    def setUp(self):
-        super().setUp()
-        self.env = self.env(user=self.env.ref('base.user_admin'))
-        archive_products(self.env)
-        account_obj = self.env['account.account']
+class TestFrontend(AccountTestInvoicingCommon, HttpCaseWithUserDemo):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+        cls.user_demo.groups_id += cls.env.ref('point_of_sale.group_pos_manager') + cls.env.ref('account.group_account_invoice')
+
+        user_admin = cls.env.ref('base.user_admin')
+        (cls.user_demo + user_admin).write({
+            'company_id': cls.env.company.id,
+            'company_ids': [(4, cls.env.company.id)],
+        })
+        cls.env = cls.env(user=user_admin)
+        archive_products(cls.env)
+        account_obj = cls.env['account.account']
 
         account_receivable = account_obj.create({'code': 'X1012',
                                                  'name': 'Account Receivable - Test',
                                                  'account_type': 'asset_receivable',
                                                  'reconcile': True})
 
-        drinks_category = self.env['pos.category'].create({'name': 'Drinks'})
+        drinks_category = cls.env['pos.category'].create({'name': 'Drinks'})
 
-        printer = self.env['pos.printer'].create({
+        printer = cls.env['pos.printer'].create({
             'name': 'Preparation Printer',
             'epson_printer_ip': '127.0.0.1',
             'printer_type': 'epson_epos',
             'product_categories_ids': [drinks_category.id]
         })
 
-        main_company = self.env.ref('base.main_company')
+        main_company = cls.env.company
 
-        second_cash_journal = self.env['account.journal'].create({
+        cls.env['pos.payment.method'].create({
+            'name': 'Bank',
+            'journal_id': cls.company_data['default_journal_bank'].id,
+            'receivable_account_id': cls.company_data['default_account_receivable'].id,
+            'company_id': cls.env.company.id,
+        })
+        second_cash_journal = cls.env['account.journal'].create({
             'name': 'Cash 2',
             'type': 'cash',
             'company_id': main_company.id
-            })
+        })
 
-        self.env['pos.payment.method'].create({
+        cls.env['pos.payment.method'].create({
             'name': 'Cash 2',
             'split_transactions': False,
             'receivable_account_id': account_receivable.id,
             'journal_id': second_cash_journal.id,
         })
 
-        pos_config = self.env['pos.config'].create({
-            'name': 'Bar',
+        pos_config = cls.env['pos.config'].create({
+            'name': 'Bar Prout',
```

Comment:
Tf ?

Reply:
It's an energy unit of measure. See: https://en.wikipedia.org/wiki/Prout_(unit)

Reply:
Some kind of "Atomic Bar", but more specialized. Good finding.